proto/public/uber/cadence/api/v1/history.proto

```diff
@@ -0,0 +1,499 @@
+    ChildWorkflowExecutionCanceledEventAttributes child_workflow_execution_canceled_event_attributes = 41;
+    ChildWorkflowExecutionTimedOutEventAttributes child_workflow_execution_timed_out_event_attributes = 42;
+    ChildWorkflowExecutionTerminatedEventAttributes child_workflow_execution_terminated_event_attributes = 43;
+    SignalExternalWorkflowExecutionInitiatedEventAttributes signal_external_workflow_execution_initiated_event_attributes = 44;
+    SignalExternalWorkflowExecutionFailedEventAttributes signal_external_workflow_execution_failed_event_attributes = 45;
+    ExternalWorkflowExecutionSignaledEventAttributes external_workflow_execution_signaled_event_attributes = 46;
+    UpsertWorkflowSearchAttributesEventAttributes upsert_workflow_search_attributes_event_attributes = 47;
+  }
+}
+
+message WorkflowExecutionStartedEventAttributes {
+  WorkflowType workflow_type = 1;
+  string parent_workflow_domain = 2;
+  WorkflowExecution parent_workflow_execution = 3;
+  int64 parent_initiated_event_id = 4;
+  TaskList task_list = 5;
+  Payload input = 6;
+  google.protobuf.Duration execution_start_to_close_timeout = 7 [(gogoproto.stdduration) = true];
+  google.protobuf.Duration task_start_to_close_timeout = 8 [(gogoproto.stdduration) = true];
+  string continued_execution_run_id = 9;
+  ContinueAsNewInitiator initiator = 10;
+  Failure continued_failure = 11;
+  Payload last_completion_result = 12;
+  // This is the runID when the WorkflowExecutionStarted event is written
+  string original_execution_run_id = 13;
+  string identity = 14;
+  // This is the very first runID along the chain of ContinueAsNew and Reset.
+  string first_execution_run_id = 15;
+  RetryPolicy retry_policy = 16;
+  int32 attempt = 17;
+  google.protobuf.Timestamp expiration_time = 19 [(gogoproto.stdtime) = true];
+  string cron_schedule = 20;
+  google.protobuf.Duration first_decision_task_backoff = 21 [(gogoproto.stdduration) = true];
+  Memo memo = 22;
+  SearchAttributes search_attributes = 23;
+  ResetPoints prev_auto_reset_points = 24;
+  Header header = 25;
+}
+
+message WorkflowExecutionCompletedEventAttributes {
+  Payload result = 1;
+  int64 decision_task_completed_event_id = 2;
+}
+
+message WorkflowExecutionFailedEventAttributes {
+  Failure failure = 1;
+  int64 decision_task_completed_event_id = 2;
+}
+
+message WorkflowExecutionTimedOutEventAttributes {
+  TimeoutType timeout_type = 1;
+}
+
+message DecisionTaskScheduledEventAttributes {
+  TaskList task_list = 1;
+  google.protobuf.Duration start_to_close_timeout = 2 [(gogoproto.stdduration) = true];
+  int32 attempt = 3;
+}
+
+message DecisionTaskStartedEventAttributes {
+  int64 scheduled_event_id = 1;
+  string identity = 2;
+  string request_id = 3;
+}
+
+message DecisionTaskCompletedEventAttributes {
+  int64 scheduled_event_id = 1;
+  int64 started_event_id = 2;
+  string identity = 3;
+  string binary_checksum = 4;
+  bytes execution_context = 5;
```

Comment:
same "maybe DataBlob" note for execution_context (also in service_worker.proto), though I'm fuzzier on this one's purpose.

Reply:
I have check [Go](https://github.com/uber-go/cadence-client/blob/4653b9fd2a2ee0b7b9b6cfd5245b3f5f0414e325/internal/internal_task_handlers.go#L1548) and [Java](https://github.com/uber/cadence-java-client/blob/5e9b33fdb0b8002c92ae50e2141e6b4d02fc74f8/src/main/java/com/uber/cadence/internal/replay/ReplayDecisionTaskHandler.java#L242) clients, and they seems not to set this field.

@meiliang86 do we still need this field? What is the purpose of it?

Reply:
@yycptt @yux0 @andrewjdawson2016 any idea?